tests/x11/user_defined_snapshot.pm

```diff
@@ -65,7 +65,12 @@ sub run() {
     save_screenshot;
     wait_screen_change { send_key 'ret' };
     # boot into the snapshot
+    # unset boot hdd image to not try to search for the disk image boot entry
+    # again as we already selected it here
+    my $boot_hdd_image = get_var('BOOT_HDD_IMAGE');
```

Comment:
do you know where inside wait_boot()  BOOT_HDD_IMAGE is used ? what about transfer this decision directly to that place so it will be clear  that here we ignoring BOOT_HDD_IMAGE ? 
By introducing such logic you sign agreement that no one will change BOOT_HDD_IMAGE inside wait_boot() and if someone will still change it will really hard to understand what's wrong , same for case when someone who didn't saw your change will try to understand what a heck is happening with wait_boot() which not boot from HDD while BOOT_HDD_IMAGE is 1